v1/python-sdk/tutorials/automl-with-azureml/forecasting-demand-forecasting-many-models/auto-ml-forecasting-demand-forecasting-many-models.ipynb

```diff
@@ -0,0 +1,1626 @@
+    "    compute_target=compute_target,\n",
+    "    node_count=5,\n",
+    "    process_count_per_node=2,\n",
+    "    run_invocation_timeout=(EXPERIMENT_TIMEOUT_HOURS * 3600 + 1000),\n",
+    "    train_pipeline_parameters=mm_paramters,\n",
+    "    arguments=[\"--retrain_failed_models\", \"True\"],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note, the output of the previous cell prints out the name of the partitioned dataset. This allows you to run a new experiment on the already partiitoned dataset. What this does is it skips a data partitioning step, and reduces the runtime. To use already partioned dataset, uncomment and execute the following code _**before**_ building the `training_pipeline_steps`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# from azureml.data.dataset_factory import TabularDatasetFactory\n",
+    "# from azureml.core.dataset import Dataset\n",
+    "\n",
+    "# PARTITIONED_TRAIN_DATASET_NAME = \"<Paste the output of the previous window here.>\"\n",
+    "# train_dataset = Dataset.get_by_name(ws, name=PARTITIONED_TRAIN_DATASET_NAME)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Submit the pipeline to run\n",
+    "Next we submit our pipeline to run. The whole training pipeline takes about 20 minutes on a 5 node STANDARD_DS15_V2  cluster."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.pipeline.core import Pipeline\n",
+    "\n",
+    "training_pipeline = Pipeline(ws, steps=training_pipeline_steps)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f\"Are many models trained? {IS_MANY_MODELS_TRAINED}\\n---\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if not IS_MANY_MODELS_TRAINED:\n",
+    "    print(\"Training new AutoML model ...\\n---\")\n",
+    "    training_run = experiment.submit(training_pipeline)\n",
+    "else:\n",
+    "    from azureml.train.automl.run import AutoMLRun\n",
+    "    from azureml.pipeline.core.run import PipelineRun\n",
+    "\n",
+    "    PIPELINE_RUN_ID = \"<Replace with the PipelineRunId used for training many models>\"  # Copy the output of Submitted PipelineRun to re-use trained models\n",
+    "    training_run = PipelineRun(experiment=experiment, run_id=PIPELINE_RUN_ID)\n",
+    "    print(f\"Using previously trained model. Pipeline run ID: {PIPELINE_RUN_ID}\\n---\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "if not IS_MANY_MODELS_TRAINED:\n",
+    "    training_run.wait_for_completion(show_output=False)\n",
```

Comment:
We can move this part to previous cell for brevity.

Reply:
done.